packages/typespec-vscode/src/emit/emitter.ts

```diff
@@ -0,0 +1,42 @@
+import vscode from "vscode";
+import { SettingName } from "../types.js";
+
+export enum EmitterKind {
+  Schema = "schema",
+  Client = "client",
+  Server = "server",
+}
+
+export interface Emitter {
+  language: string;
+  package: string;
+  version?: string;
+  kind: EmitterKind;
+}
+
+const extensionConfig = vscode.workspace.getConfiguration();
+
+function getEmitter(kind: EmitterKind, emitter: Emitter): Emitter {
+  const packageFullName: string = emitter.package ?? "";
```

Comment:
what would happen if the package is "" or undefined? will it show an empty item in the UI? if so it's better to filter them out and log a warning for them.